### PR TITLE
582864 upgrade to spring 6 and osgi 7 and several unit tests to junit 4

### PR DESCRIPTION
**Pull request by mxcross**
upgrade to spring 6 and osgi 7 and several unit tests to junit 4

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -23,29 +23,31 @@
...
-        <slf4j.version>1.7.5</slf4j.version>
-        <logback.version>1.0.13</logback.version>
+        <slf4j.version>2.0.12</slf4j.version>
+        <logback.version>1.5.3</logback.version>
         <env.buildPlan />
         <env.buildNumber />
 
         <!-- compiled against osgi Revision -->
-        <osgi.ver>5.0.0</osgi.ver>
+        <osgi.ver>7.0.0</osgi.ver>
```
Comment (kennymacleod) [inline pom.xml]: why not 8.0.0?

**Comment by wilx** [comment]
How does this differ from #11, #12, #13? And there is also #10.

**Comment by wilx** [comment]
I am not a boss here but please, can you guys combine your work and get this new version rolling?

**Comment by mxcross** [comment]
> How does this differ from #11, #12, #13? And there is also #10.

#11, #12, and #13 are all spring 6.0, not 6.1.

#10 updates the spring library but does not have the changes to make the code or tests work. It actually is the branch I started with to make the changes, but it was done on a non-public github.

File: pom.xml (hunk 1 above)
Comment (mxcross) [reply pom.xml]: 7.0.0 is the minimum version needed to make it compatible with the felix and equinox updates

**Comment by wilx** [comment]
I believe the changes from #16 will be necessary as well for this to work.

**Comment by mxcross** [comment]
> I believe the changes from #16 will be necessary as well for this to work.

Thank you for the update! I did include it, but I'm still getting a failure on the continuous-integration step. It seems it's not using the update to the Jenkinsfile because I'm not considered trusted.

**Comment by wilx** [comment]
> > I believe the changes from #16 will be necessary as well for this to work.
> 
> Thank you for the update! I did include it, but I'm still getting a failure on the continuous-integration step. It seems it's not using the update to the Jenkinsfile because I'm not considered trusted.

Neither am I. I am not sure what to do about it. @olaf-otto?

**Comment by wilx** [comment]
Are these necessary in compile scope? It seems to me they should be at best test dependencies:
https://github.com/mxcross/gemini.blueprint/blob/acac7204e1bb2b5069dde34459b7f896ab1fd442/pom.xml#L358-L384

**Comment by mxcross** [comment]
> Are these necessary in compile scope? It seems to me they should be at best test dependencies: https://github.com/mxcross/gemini.blueprint/blob/acac7204e1bb2b5069dde34459b7f896ab1fd442/pom.xml#L358-L384

those dependencies are for logging as the comment suggests in the pom.

**Comment by wilx** [comment]
No part of this project is using SLF4J or Logback. The commons-logging `Log` class can be provided either by `spring-jcl` or by `commons-logging`.

**Comment by mxcross** [comment]
> No part of this project is using SLF4J or Logback. The commons-logging `Log` class can be provided either by `spring-jcl` or by `commons-logging`.

although no references are in the java code, there are references in the bundle configurations. I'd rather not remove dependencies I didn't add that would impact logging.

**Comment by olaf-otto** [comment]
> > > I believe the changes from #16 will be necessary as well for this to work.
> > 
> > 
> > Thank you for the update! I did include it, but I'm still getting a failure on the continuous-integration step. It seems it's not using the update to the Jenkinsfile because I'm not considered trusted.
> 
> Neither am I. I am not sure what to do about it. @olaf-otto?

Hi! I have access to Jenkins, not sure what the message "‘Jenkinsfile’ has been modified in an untrusted revision" will yield, but it looks like a warning. The actual issue is:

````
WorkflowScript: 47: Duplicate stage name: "Build with Equinox profile" @ line 47, column 3.
     stages {
````

@mxcross I will try and apply your changes from the Jenkinsfile on main directly.

**Comment by olaf-otto** [comment]
@mxcross I've applied the changes to main - can you please modify the PR accordingly? Thanks!

**Comment by mxcross** [comment]
> @mxcross I've applied the changes to main - can you please modify the PR accordingly? Thanks!

Updated. Thank you!

**Comment by mxcross** [comment]
@olaf-otto I'm finding that even on the main branch that the integration-tests folder doesn't compile nor does it pass if you go through the trouble of getting them to compile. Is there some plan around these since they're gating any PRs from the repo?

**Comment by olaf-otto** [comment]
> @olaf-otto I'm finding that even on the main branch that the integration-tests folder doesn't compile nor does it pass if you go through the trouble of getting them to compile. Is there some plan around these since they're gating any PRs from the repo?

Oh, I was unaware of that situation. I shall take a look tonight, it should compile and pass indeed. Thanks for pointing that out!

**Comment by olaf-otto** [comment]
Hi @mxcross I have fixed the integration tests on the main branch. There, I reverted to JDK 8 since the main branch is still on that language level. I have also made sure that clean integration testing cycles are run for each tested OSGi container.

Thus, the expectation is that the integration test shall also pass after the upgrade. Migrating the integration tests accordingly might pose a challenge, since they are sophisticated and  neither well documented nor easily readable.

**Comment by mxcross** [comment]
@olaf-otto thank you for fixing the integration tests on main. I agree with your assessment that it will be challenging. At least now there's something that can be run through a debugger that works. Much appreciated!

**Comment by wilx** [comment]
Ping. How far is this project?

**Comment by mxcross** [comment]
The lift to get the integration tests running/passing is greater than the upgrade itself.

**Comment by AxeTheAxe** [comment]
Hi @mxcross ! What was the reason to close this PR without merging? The problems with the IT?

**Comment by mxcross** [comment]
@AxeTheAxe Yup, the IT's weren't even in a compiling state and they are extremely complex. The lift to get those working again seems larger than the upgrade itself.

**Comment by garpinc** [comment]
So which branch is the one we should be using? the eclipse-gemini:feature/582864-upgrade-to-spring-6 branch or this one? I Looks like this is the only one where all the PRs are merged in right? I'm on spring 6.1.6.1. 

Will I be able to get a build that works even though the IT tests don't? Are there binary jars somewhere so I don't have to fight through the compilation?

**Comment by mxcross** [comment]
> So which branch is the one we should be using? the eclipse-gemini:feature/582864-upgrade-to-spring-6 branch or this one? I Looks like this is the only one where all the PRs are merged in right? I'm on spring 6.1.6.1.
> 
> Will I be able to get a build that works even though the IT tests don't? Are there binary jars somewhere so I don't have to fight through the compilation?

As long as you aren't using the "it" profile it should compile for you fine. It compiles, unit tests pass. I did have it functional locally.

**Comment by arun-madan-1** [comment]
@mxcross I noticed a new version of Gemini Blueprint (4.0.0-atlassian-4) available on maven central which supports spring 6. Do you have any idea if its good to use or not?

**Pull request closed without merging**